### PR TITLE
Allow cross-platform installations

### LINKED ISSUE #23: Unable to install for other platforms
When installing for a different platform (or cross-compiling with conda-build) installing the environment fails:

```bash
$ CONDA_SUBDIR=linux-ppc64le conda create --name test gdk-pixbuf
.....
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
ERROR conda.core.link:_execute(700): An error occurred while installing package 'conda-forge::gdk-pixbuf-2.38.2-hb6602f3_4'.
Rolling back transaction: done

LinkError: post-link script failed for package conda-forge::gdk-pixbuf-2.38.2-hb6602f3_4
location of failed script: /home/cburr/miniconda3/envs/test/bin/.gdk-pixbuf-post-link.sh
==> script messages <==
<None>
==> script output <==
stdout:
stderr: /home/cburr/miniconda3/envs/test/bin/.gdk-pixbuf-post-link.sh: line 21: /home/cburr/miniconda3/envs/test/bin/gdk-pixbuf-query-loaders: cannot execute binary file: Exec format error

return code: 126

()

```

This is due to the post link script and to avoid this the [conda-build docs state](https://conda.io/projects/conda-build/en/latest/resources/link-scripts.html):

> Post-link and pre-unlink scripts should:
>  * Depend only on simple system tools such as rm, cp, mv, and ln.

I've not looked into the details yet but is there any way running `gdk-pixbuf-query-loaders` can be avoided?

### PR DESCRIPTION
**Pull request by chrisburr**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Fixes #23

### CODE REVIEW

**Comment by chrisburr** [comment]
@conda-forge-admin, please rerender

**Comment by chrisburr** [comment]
I've ran into an issue that the Python version of `meson` from `build` can differ from the one in `host`. I don't quite understand what `gobject-introspection` is all about but it seems odd (e.g. python bindings are in `lib/gobject-introspection` instead of `site-packages`). Maybe it belongs as a `build:` dependency?

If not, I think the linter can be ignored for now. Once `gobject-introspection` is updated for Python 3.9 the problem will go away for another year.

**Comment by pkgw** [comment]
@chrisburr Yeah, gobject-introspection is a bit funky. Part of the issue is that it could/should probably be split into sub-packages, since it's combining a few different pieces of functionality. [This page](https://gi.readthedocs.io/en/latest/) gives a nice overview. The G-I Python code powers the build-time scanner and compiler tools. The Python code is considered an implementation detail, not a Python package that's meant to be consumed by others, which is why it's not in site-packages.

Assuming that we're not going to go ahead and split g-i in the very near future, your workaround is probably fine for now.

File: recipe/post-link.sh [hunk 1]
```diff
@@ -9,13 +9,18 @@
...
+    if [ ! -d "$bindir" ] ; then
        echo >&2 "error: no such gdk-pixbuf binary directory $PREFIX/lib/gdk-pixbuf-2.0/*"
        exit 1
     fi
     gdk_pixbuf_cache_file="$bindir/loaders.cache"
 fi
-"$PREFIX/bin/gdk-pixbuf-query-loaders" >$gdk_pixbuf_cache_file
+"$PREFIX/bin/gdk-pixbuf-query-loaders" >"$gdk_pixbuf_cache_file" 2>>"${PREFIX}/.messages.txt" || \
+(
+    echo "ERROR: Failed to update gdk-pixbuf's cache, some plugins may be be found."
```
Comment (pkgw) [inline recipe/post-link.sh:23]: Confusing typo: "be be".

Please also add a source code comment mentioning that one situation in which the command might fail is if the package is being installed into an environment for a foreign architecture (though I'm not sure if my terminology is fully right for this).

**Pull request merged**